File: ShippingMark/Models/ThongBaoXuatHang.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ShippingMark.Models
{
    public class ThongBaoXuatHang
    {
        // --- BƯỚC 1: Thêm Constructor để khởi tạo các danh sách ---
        public ThongBaoXuatHang()
        {
            this.Containers = new HashSet<Container>();
            this.ChiTietHangHoas = new HashSet<ChiTietHangHoa>();
        }

        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ID { get; set; }

        [Display(Name = "Loại hàng")]
        public string LoaiHang { get; set; }

        [Display(Name = "Số lượng")]
        public int SoLuong { get; set; }

        [Display(Name = "REFNO Đơn hàng")]
        public string REFNO { get; set; }

        [Display(Name = "PoNo Số PO")]
        public string PoNo { get; set; }

        [Required(ErrorMessage = "Vui lòng chọn ngày dự kiến.")]
        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        [Display(Name = "Ngày dự kiến")]
        public DateTime? Ngaydukien { get; set; }

        [Required(ErrorMessage = "Vui lòng chọn ngày ETD.")]
        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        [Display(Name = "Ngày ETD")]
        public DateTime? NgayETD { get; set; }

        [Display(Name = "Ghi chú")]
        public string GhiChu { get; set; }

        // --- Liên kết đến các bảng khác ---
        [ForeignKey("KhachHang")]
        public int? KhachHang_ID { get; set; }
        public virtual KhachHang KhachHang { get; set; }

        [ForeignKey("HangHoa")]
        public int? HangHoa_ID { get; set; }
        public virtual HangHoa HangHoa { get; set; }
        // === BẮT ĐẦU NÂNG CẤP ===
        // Lưu thông tin người tạo và thời gian tạo
        [ForeignKey("CreatedByUser")]
        public string CreatedById { get; set; }
        public virtual ApplicationUser CreatedByUser { get; set; }
        public DateTime CreatedDate { get; set; }

        // Lưu thông tin người sửa cuối cùng và thời gian sửa
        [ForeignKey("LastModifiedByUser")]
        public string LastModifiedById { get; set; }
        public virtual ApplicationUser LastModifiedByUser { get; set; }
        public DateTime? LastModifiedDate { get; set; }
        // === KẾT THÚC NÂNG CẤP ===

        // --- Các thuộc tính Collection ---
        public virtual ICollection<Container> Containers { get; set; }
        public virtual ICollection<ChiTietHangHoa> ChiTietHangHoas { get; set; }
    }
}

File: app/Models/ThongBaoXuatHang.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ThongBaoXuatHang extends Model
{
    use HasFactory;

    protected $table = 'thong_bao_xuat_hangs';

    protected $fillable = [
        'loai_hang',
        'so_luong',
        'ref_no',
        'po_no',
        'ngay_du_kien',
        'ngay_etd',
        'ghi_chu',
        'khach_hang_id',
        'hang_hoa_id',
        'created_by',
        'last_modified_by',
    ];

    protected $casts = [
        'ngay_du_kien' => 'date',
        'ngay_etd' => 'date',
        'so_luong' => 'integer',
    ];

    public function khachHang()
    {
        return $this->belongsTo(KhachHang::class, 'khach_hang_id');
    }

    public function hangHoa()
    {
        return $this->belongsTo(HangHoa::class, 'hang_hoa_id');
    }

    public function chiTietHangHoas()
    {
        return $this->hasMany(ChiTietHangHoa::class, 'thong_bao_xuat_hang_id');
    }

    public function containers()
    {
        return $this->hasMany(Container::class, 'thong_bao_xuat_hang_id');
    }

    public function creator()
    {
        return $this->belongsTo(User::class, 'created_by');
    }

    public function editor()
    {
        return $this->belongsTo(User::class, 'last_modified_by');
    }
}
